packages/notebook/src/widget.ts

```diff
@@ -1274,6 +1282,32 @@ export class Notebook extends StaticNotebook {
     this.addFooter();
   }
 
+  /**
+   * Access previous item in ipython kernel for current kernel
+   */
+  accessLastHistory(): void {
+    const activeCell = this.activeCell;
+    if (activeCell) {
+      if (this._history) {
+        const lastHistory = this._history.back(activeCell);
+        this._history.updateEditor(activeCell, lastHistory);
```

Comment:
I would suggest implementing this as async rather than passing a promise.

```suggestion
        const lastHistory = await this._history.back(activeCell);
        this._history.updateEditor(activeCell, lastHistory);
```

Reply:
Does it make sense to not have them be promises or async at all. As far as I can tell all the calls within the function are synchronous so maybe it makes sense to just return a string. with no await.